core/src/main/java/org/mskcc/oncotree/topbraid/TopBraidSessionConfiguration.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2017 Memorial Sloan-Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE.  The software and
+ * documentation provided hereunder is on an "as is" basis, and
+ * Memorial Sloan-Kettering Cancer Center
+ * has no obligations to provide maintenance, support,
+ * updates, enhancements or modifications.  In no event shall
+ * Memorial Sloan-Kettering Cancer Center
+ * be liable to any party for direct, indirect, special,
+ * incidental or consequential damages, including lost profits, arising
+ * out of the use of this software and its documentation, even if
+ * Memorial Sloan-Kettering Cancer Center
+ * has been advised of the possibility of such damage.
+*/
+
+package org.mskcc.oncotree.topbraid;
+
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.log4j.Logger;
+import org.apache.http.client.CookieStore;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.StatusLine;
+import org.apache.http.client.methods.HttpHead;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.cookie.Cookie;
+import org.apache.http.impl.cookie.BasicClientCookie;
+import org.apache.http.impl.client.BasicCookieStore;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpStatus;
+
+/**
+ *
+ * @author Manda Wilson
+ **/
+@Configuration
+public class TopBraidSessionConfiguration {
+
+    private final static Logger logger = Logger.getLogger(TopBraidSessionConfiguration.class);
+
+    @Value("${topbraid.url}")
+    private String topBraidURL;
+    
+    public String getSessionId(String username, String password) {
```

Comment:
This is because the URL is general to TopBraid and the username and password is specific to Oncotree.  The OncotreeRepository has the username and parameter set by Spring and passes it to the parent TopBraidRepository.  You had said we might have different usernames and passwords for different things within TopBraid.  This is sort of unnecessary since this is the Oncotree app, rather than the planned generic TopBraid app, but since the code is already designed to plugin another [NewType]Repository, I would leave it ready to do that (ready to be pulled out into a separate web API).